inference-engine/src/transformations/src/transformations/convert_quantize_dequantize.cpp

```diff
@@ -0,0 +1,139 @@
+//                              +---------------------+
+//                              |      Convert        |
+//                              |(e.g. from f32 to u8)|
+//                              +---------+-----------+                            |  |  |  |  |
+//                                        |                                        |  |  |  |  |
+//                                        v                                        v  v  v  v  v
+//                              +---------------------+                           +------------+
+//                              |      Convert        |            ====>          |FakeQuantize|
+//                              |  (from u8 to f32)   |                           +------------+
+//                              +---------+-----------+                                 |
+//                                        |                                             v
+//                                        v
+//                  +----------+    +------------+
+//                  |zero point|--->|  Subtract  |
+//                  +----------+    +-----+------+
+//                                        |
+//                                        v
+//                   +---------+    +------------+
+//                   |  scale  |--->|  Multiply  |
+//                   +---------+    +-----+------+
+//                                        |
+//                                        v
+//
+
+
+ngraph::pass::ConvertQuantizeDequantize::ConvertQuantizeDequantize() {
+    auto data_pattern = ngraph::pattern::any_input();
+    auto input_low_pattern = ngraph::pattern::any_input();
+    auto input_high_pattern = ngraph::pattern::any_input();
+    auto output_low_pattern = ngraph::pattern::wrap_type<opset1::Constant>();
+    auto output_high_pattern = ngraph::pattern::wrap_type<opset1::Constant>();
+    auto fq_pattern = ngraph::pattern::wrap_type<opset1::FakeQuantize>({data_pattern, input_low_pattern,
+                                                                       input_high_pattern, output_low_pattern,
+                                                                       output_high_pattern});
+    auto convert1_pattern = ngraph::pattern::wrap_type<opset1::Convert>({fq_pattern}, pattern::type_matches_any({element::i8, element::u8}));
+    auto convert2_pattern = ngraph::pattern::wrap_type<opset1::Convert>({convert1_pattern}, pattern::type_matches(element::f32));
+    auto zero_point_pattern = ngraph::pattern::any_input();
+    auto sub_pattern = ngraph::pattern::wrap_type<opset1::Subtract>({convert2_pattern, zero_point_pattern}, pattern::consumers_count(1));
+    auto scale_pattern = ngraph::pattern::any_input();
+    auto mul_pattern = ngraph::pattern::wrap_type<opset1::Multiply>({sub_pattern, scale_pattern});
+
+    ngraph::graph_rewrite_callback callback = [=](pattern::Matcher& m) {
+        auto pattern_map = m.get_pattern_map();
```

Comment:
You need to use `m.get_pattern_value_map()` in case if you get matched outputs and use them as inputs for other operations. 

For example if `input_low` is a node with more than one output and you use pattern_map ( pattern_map[input_low_pattern]; will return node pointer but not output) then if you pass this node as input to another node then it will lead to an undefined behavior because default 0 output will be used. So to avouid this bad situation you need to use `pattern_value_map`